### PR TITLE
David's emacs tweaks

### PR DESCRIPTION
**Pull request by david-christiansen**
Here's a few updates to today's Emacs mode:

1. Various Emacs package conventions are implemented, preparing for submission to MELPA
2. Highlighting for atoms works
3. Various bugfixes, such as in `imenu` and the autoload cookies
4. Minor aesthetic tweaks, such as calling the compilation buffer `*RedPRL*` instead of `*redprl*`
5. The mode hook is added to Customize
6. Starting `redprl-mode` will optionally display a revolutionary slogan in the echo area

### CODE REVIEW

**Comment by david-christiansen** [comment]
Feel free to cherry-pick if some of these are unwelcome.

**Comment by david-christiansen** [comment]
I suppose this is all RE #50

File: emacs/redprl.el [hunk 1]
```diff
@@ -164,11 +164,11 @@
...
-       `(("Def" (rx-to-string ,redprl-def-name-regexp) 1)
-         ("Thm" (rx-to-string ,redprl-thm-name-regexp) 1)
-         ("Tac" (rx-to-string ,redprl-tac-name-regexp) 1)
-         ("Sym" (rx-to-string ,redprl-sym-name-regexp) 1)
-         ("Record" (rx-to-string ,redprl-record-name-regexp) 1)))
+       `(("Def" ,(rx-to-string redprl-def-name-regexp) 1)
+         ("Thm" ,(rx-to-string redprl-thm-name-regexp) 1)
+         ("Tac" ,(rx-to-string redprl-tac-name-regexp) 1)
+         ("Sym" ,(rx-to-string redprl-sym-name-regexp) 1)
+         ("Record" ,(rx-to-string redprl-record-name-regexp) 1)))
```
Comment (jonsterling) [inline emacs/redprl.el]: thank you for fixing this! When it was "working", it must have just had the version loaded from before I broke this.

**Comment by jonsterling** [comment]
This looks great, thanks! Thanks also for the granular commits, so that I can learn about each change.

**Comment by david-christiansen** [comment]
Cheers! Want me to submit to MELPA?

**Comment by david-christiansen** [comment]
(we can point the MELPA version at this repo and tell it to build the elisp file, so there's no extra synch overhead)

**Comment by jonsterling** [comment]
@david-christiansen sure! sounds great!

**Pull request merged**